types/persisted-model.d.ts

```diff
@@ -478,58 +477,6 @@ export declare class PersistedModel extends ModelBase {
     callback?: Callback<boolean>,
   ): PromiseOrVoid<boolean>;
 
-/**
- * Begin a new transaction.
- *
- * A transaction can be committed or rolled back. If timeout happens, the
- * transaction will be rolled back. Please note a transaction is typically
- * associated with a pooled connection. Committing or rolling back a transaction
- * will release the connection back to the pool.
- *
- * Once the transaction is committed or rolled back, the connection property
- * will be set to null to mark the transaction to be inactive. Trying to commit
- * or rollback an inactive transaction will receive an error from the callback.
- *
- * Please also note that the transaction is only honored with the same data
- * source/connector instance. CRUD methods will not join the current transaction
- * if its model is not attached the same data source.
- *
- * Example:
- *
- * To pass the transaction context to one of the CRUD methods, use the `options`
- * argument with `transaction` property, for example,
- *
- * ```js
- * MyModel.beginTransaction('READ COMMITTED', function(err, tx) {
-  *   MyModel.create({x: 1, y: 'a'}, {transaction: tx}, function(err, inst) {
-  *     MyModel.find({x: 1}, {transaction: tx}, function(err, results) {
-  *       // ...
-  *       tx.commit(function(err) {...});
-  *     });
-  *   });
-  * });
-  * ```
-  *
-  * @param {Object|String} options Options to be passed upon transaction.
-  *
-  * Can be one of the forms:
-  *  - Object: {isolationLevel: '...', timeout: 1000}
-  *  - String: isolationLevel
-  *
-  * Valid values of `isolationLevel` are:
-  *
-  * - Transaction.READ_COMMITTED = 'READ COMMITTED'; // default
-  * - Transaction.READ_UNCOMMITTED = 'READ UNCOMMITTED';
-  * - Transaction.SERIALIZABLE = 'SERIALIZABLE';
-  * - Transaction.REPEATABLE_READ = 'REPEATABLE READ';
-  * @callback {Function} cb Callback function.
-  * @returns {Promise|undefined} Returns a callback promise.
-  */
-  beginTransaction(
```

Comment:
I have a question. Does `beginTransaction` exist in `PersistedModel` constructor or not? If it exists, then I think we should keep this definition, as removing it could be considered a breaking change. If it does not exist, then I am fine to remove this definition because it's not working anyways.

Reply:
I added this change in https://github.com/strongloop/loopback-datasource-juggler/pull/1752 and no it doesn't (see https://github.com/strongloop/loopback/blob/master/lib/persisted-model.js), which is why I'm reverting the changes made in that PR. IIUC it exists on a persisted model once an application is booted and `TransactionMixin` is applied.

Reply:
It seems that `beginTransaction` is added to models when they are attached to a (compatible) data-source. 

Initial state - `beginTransaction` is not available:

```
$ node
const {PersistedModel} = require('loopback');
undefined
> PersistedModel.beginTransaction
undefined
```

Using https://github.com/strongloop/loopback-sandbox, adding `console.log` statement after the app has booted (https://github.com/strongloop/loopback-sandbox/blob/96e99c26ae7ff07ca0d97c073b795bb6fcdf987b/server/server.js#L28):

```js
$ node .
[Function]
Web server listening at: http://localhost:3000
Browse your REST API at http://localhost:3000/explorer
```

Maybe we should keep `beginTransaction` but make it optional?

```suggestion
  beginTransaction?(
```

@raymondfeng what's your take?

I don't have a strong opinion, I am ok to remove this recently added typedef.

Reply:
Thank you for investigating @bajtos. Yea we can definitely do that - I don't have a strong opinion either. If we remove it from the declaration file does that constitute a breaking change? I.e. what do you think the semver version change should be depending on removal or making it optional?

Reply:
Let's remove it as a decision to not expose it to TS.